include/domain/impl/receivers/kernel.tpp

```diff
@@ -186,10 +194,110 @@ void specfem::domain::impl::kernels::receiver_kernel<
                                  point_properties,
```

Comment:
I think it would be ideal if we can pass seismogram type to this function. I realize for pressure we need the seismo type within `get_field`. With your current formulation this seems not possible, since we need the divergence. Can you look at the SPECFEM2D formulation https://github.com/SPECFEM/specfem2d/blob/master/src/specfem2D/compute_pressure.f90 .

With that formulation we should be able to compute pressure using only gradients